src/mkdocs_autorefs/plugin.py

```diff
@@ -133,7 +137,14 @@ def on_config(self, config: MkDocsConfig) -> MkDocsConfig | None:
             The modified config.
         """
         log.debug("Adding AutorefsExtension to the list")
-        config["markdown_extensions"].append(AutorefsExtension())
+        for ext in config.markdown_extensions:
+            if ext == "attr_list" or isinstance(ext, AttrListExtension):
+                log.debug("Enabling Markdown anchors feature")
+                scan_anchors = True
```

Comment:
https://github.com/mkdocstrings/mkdocstrings/blob/f5536840902883faabf8d2d9d7f04d8452dc63b2/src/mkdocstrings/plugin.py#L184
mkdocstrings sets `.scan_toc = False` but actually the intention of that was "subdue all of the plugin's own activity" and I do believe that this new behavior should also be subdued. To not cause any accidents, let's keep  using `.scan_toc = False` to mean "disable all scanning", including this new feature.

I propose a change accordingly + a refactor to detect the extension via the treeprocessor registry which seems simpler.
https://github.com/oprypin/mkdocs-autorefs/commit/cc7ab04767ba46145a1cd0d9695d9badb32c4bf4#diff-30fb6dd2364200c9fce03f30c248a8eba46fd7b4c733fe9c10ab6db1e7f35ac7